hedera-node/hapi-utils/src/test/java/com/hedera/node/app/hapi/utils/CommonUtilsTest.java

```diff
@@ -111,5 +208,112 @@ void detectsOverflowInVariousCases() {
         assertTrue(productWouldOverflow(overflowMultiplier, nonZeroMultiplicand));
     }
 
+    @Test
+    void throwsOnUnexpectedFunctionality() {
+        assertThrows(UnknownHederaFunctionality.class, () -> functionOf(TransactionBody.getDefaultInstance()));
+    }
+
+    @Test
+    void getsExpectedTxnFunctionality() {
+        final Map<HederaFunctionality, BodySetter<? extends GeneratedMessageV3, Builder>> setters = new HashMap<>() {
```

Comment:
Have you considered writing a data-driven test (one that uses `@ParameterizedTest`) instead? Creating such a data structure and iterating over it looks unusual.